File: sorting-algorithms/insertion-sort/InsertionSort.java
import java.util.Arrays;

public class InsertionSort {

    /**
     * 
     * @param data
     * @return data
     */
    public int[] sort(int data[]) {

        for (int i = 0; i < data.length; i++) {

            int current = data[i];

            int j = i - 1;

            while (j >= 0 && data[j] > current) {
                data[j + 1] = data[j];
                j = j - 1;
            }

            data[j + 1] = current;
        }
        return data;
    }

    /**
     * @param args
     */
    public static void main(String args[]) {

        InsertionSort insertionSortObj = new InsertionSort();
        int data[] = { 10, 5, 6, 9, 8, 2, 2, 11, 7 };
        int[] result = insertionSortObj.sort(data);
        System.out.println(Arrays.toString(result));
    }
}

File: sorting-algorithms/insertion-sort/InsertionSort.php
<?php




/**
 * Insertion sort function
 * complexity O(n2)
 *
 * @param array $data
 * @return array
 */
function insertionSort(array $data): array
{

    for ($i = 0; $i < count($data); $i++) {
        $current = $data[$i];

        $j = $i - 1;

        while ($j >= 0 && $data[$j] > $current) {
            $data[$j + 1] = $data[$j];
            $j = $j - 1;
        }
        $data[$j + 1] = $current;
    }

    return $data;
}





$data = [10, 5, 6, 9, 8, 2, 2, 11, 7];
$result = insertionSort($data);
print_r($result);
